Feature/ci cd to curseforge

bison92 opened the pull request:
This workflow has 1 job of 5 steps:

1. Checkout the repository
2. Create addon.zip Zip file containing:
    - Data.lua 
    - FlightTimerClassic.lua 
    - FlightTimerClassic.toc 
3. Upload zip file as an artifact, so the generated zip file can be downloaded from the pipeline, for inspection.
4. Echo the curl command that we're going to issue to the curse forge API, with the secrets obfuscated.
5. Run the curl command.

For this to work, you need to create an API token from https://www.curseforge.com/account/api-tokens. Then create 2 repository secrets (from the repository settings menu in github):

1. CURSEFORGE__TOKEN with the generated token
2. CURSEFORGE__PROJECTID this is not really a secret but this way it was easier for me to try the workflow out in a test project. It needs to be the projectid 554379 no quotes.

This will create a file called addon.zip in the project, which I'm not sure is what we're after. I was thinking we could use something like GitVersion https://gitversion.net/docs/learn/how-it-works to tag the current version with git so we could get something like FlightTimerClassic-X.Y.Z.zip instead

Commit:
add workflow

Commit:
don't upload when running for PR

Commit:
remove unnecessary envs

timohausmann commented:
Thank you so much! I will push version tags later today. I think the zip filename is visible on curse/wowup so it would be great to include the version.

The current release is already „beta“ so can we replace alpha with that or the keyword for release?

It would also be cool to keep track of the changes, could it maybe read out the detailed commit description of the tagged commit? Or I could create a json file with `version: changelog` if that helps.

I have some horde data coming and plan to do a bugfix so we can try it soon :-)

bison92 commented:
Okey, If we start tagging the commit we could extract the version out of it. 
About the changelog, we can create a file with a certain name that includes the version and read out of it or something like that.

Commit:
mark version as Beta instead of Alpha in curseforge upload

timohausmann commented:
Okay, maybe simply a file called `changelog` that will only contain plaintext for the current version?

bison92 commented:
While that could certainly work, you might want to have all the change logs there, we can specify certain format and make the script capable of reading it. Something like :

5.0.1:

- added some feature
- fixed some problem

5.0.0:

- added new settings UI

If the format is respected a simple AWK script can parse it and extract the interesting bits for the curseforge request

bison92 commented:
There's also the Github releases feature. Where you can create releases for the repository and specify the changelog https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository

timohausmann commented:
Okay, as a test I manually created a release for the current version (I also added tags for the past versions).

I had another thought that not publishing every git commit to CurseForge, but releases/tagged versions make sense. 

Could you use releases as a base for packaging? But also we don't have to overengineer this 😆
